src/Chainweb/Miner/Coordinator.hs

```diff
@@ -90,7 +91,7 @@ import Utils.Logging.Trace (trace)
 -- made as often as desired, without clogging the Pact queue.
 --
 newtype PrimedWork =
-    PrimedWork (HM.HashMap MinerId (HM.HashMap ChainId (Maybe CachedPayload)))
+    PrimedWork (HM.HashMap MinerId (HM.HashMap ChainId (TVar (Maybe CachedPayload))))
```

Comment:
Each leaf is now a `TVar`, so that the producer and consumer threads can each have fine-grained access to these chain-specific updates.